contracts/contracts/Multisig.cairo

```diff
@@ -314,14 +314,8 @@ end
 func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
     owners_len : felt, owners : felt*, confirmations_required : felt
 ):
-    const lower_bound = 1
-
-    with_attr error_message("range conditions unsatisfied"):
-        assert_in_range(confirmations_required, lower_bound, owners_len + 1)
-    end
-
-    _set_owners_impl(owners_len, owners)
-    _set_confirmations_required_impl(confirmations_required)
+    _validate_and_set_confirmations_required(confirmations_required, owners_len)
+    _set_owners(owners_len, owners)
```

Comment:
`_validate_and_set_confirmations_required` also validates `owners_len` so it makes sense to call it first

Reply:
Please switch the order of these calls so they are consistent throughout the contract.

1. Constructor should set the owners first, then threshold (currently opposite)
2. `set_owners` fn should set the owners first, update the threshold if new owners are less than the existing threshold (currently ok)
3. `set_owners_and_threshold` fn should set owners first, update the threshold second (currently opposite)

Reply:
> Constructor should set the owners first, then threshold (currently opposite)

Doesn't it make sense to first check that `owners_len` and `confirmations_required` satisfy following requirements:
`owners_len` not zero
`confirmations_required` is within [1, owners_len]. Because otherwise there is no point to proceed with setting threshold or owners
And than call set_owners which checks uniqueness owners and so on?

> set_owners_and_threshold fn should set owners first, update the threshold second (currently opposite)

Same as above

Reply:
Well [if this ](https://github.com/eqlabs/starknet-multisig/pull/22#discussion_r916683876) is not a case then swapping places is ok

Reply:
It makes sense to save some gas like you are suggesting.

> Following order can be achieved only by splitting _validate_and_set_confirmations_required on 2 functions: validate & set

IMHO this is the way to go.

1. Remove the `_validate_and_set_confirmations_required` fn altogether
2. Replace it with a standard require fn, for example `require_confirmations_required_valid` + existing setter
3. In constructor and `set_owners_and_threshold` functions validate the requirement first, then set owners and threshold

```cairo
require_confirmations_required_valid(confirmations_required, owners_len)
_set_owners(owners_len, owners)
_set_confirmations_required(confirmations_required)
```

Reply:
Great! Will do

Reply:
Done